src/npctalk.cpp

```diff
@@ -2792,6 +3193,151 @@ void talk_function::give_equipment(npc *p)
     p->add_effect("asked_for_item", 1800);
 }
 
+void talk_function::give_aid(npc *p)
+{
+    g->u.cash -= 20000;
+    p->add_effect("currently_busy", 300);
+    body_part bp_healed;
+    for (int i = 0; i < num_hp_parts; i++) {
+        bp_healed = player::hp_to_bp( hp_part(i) );
+        g->u.heal(hp_part(i), 5*rng(2,5));
+        if (g->u.has_effect("bite", bp_healed)) {
+            g->u.remove_effect("bite", bp_healed);
+        }
+        if (g->u.has_effect("bleed", bp_healed)) {
+            g->u.remove_effect("bleed", bp_healed);
+        }
+        if (g->u.has_effect("infected", bp_healed)) {
+            g->u.remove_effect("infected", bp_healed);
+        }
+    }
+    g->u.assign_activity(ACT_WAIT, 10000);
+}
+
+void talk_function::give_all_aid(npc *p)
+{
+    g->u.cash -= 30000;
+    p->add_effect("currently_busy", 300);
+    give_aid(p);
+    body_part bp_healed;
+    for( auto &elem : g->active_npc ) {
+        if (rl_dist( elem->pos(), g->u.pos() ) < PICKUP_RANGE && elem->is_friend()){
+            for (int i = 0; i < num_hp_parts; i++) {
+                bp_healed = player::hp_to_bp( hp_part(i) );
+                elem->heal(hp_part(i), 5*rng(2,5));
+                if (elem->has_effect("bite", bp_healed)) {
+                    elem->remove_effect("bite", bp_healed);
+                }
+                if (elem->has_effect("bleed", bp_healed)) {
+                    elem->remove_effect("bleed", bp_healed);
+                }
+                if (elem->has_effect("infected", bp_healed)) {
+                    elem->remove_effect("infected", bp_healed);
+                }
+            }
+        }
+    }
+}
+
+void talk_function::construction_tips(npc *p)
+{
+    g->u.cash -= 2000;
+    g->u.practice("carpentry", 30);
+    g->u.assign_activity(ACT_WAIT, 600);
+    p->add_effect("currently_busy", 600);
+}
+
+void talk_function::buy_beer(npc *p)
+{
+    item cont = item("bottle_glass", 0, false);
+    item liq = item("beer", 0);
```

Comment:
You could use `liq = liq.in_its_container();` here to put the liquid into its default container. `liq` will become a `bottle with beer` or so. The container item is taken from the item data of the liquid and it's automatically filled to the brim (the charges of the liquid are adjusted to match the container size).

This allows you to completely remove the `cont` item, the charge setting and the call to `handle_liquid` (which creates the message you complained about).